src/stores/authenticationStore.ts

```diff
@@ -0,0 +1,55 @@
+import { AccountTypes } from '@bcrs-shared-components/enums'
+import { RootStateIF } from '@/interfaces'
+import { defineStore } from 'pinia'
+
+export const useAuthenticationStore = defineStore('authentication', {
+  getters: {
+    /** Whether the current account is a premium account. */
+    isPremiumAccount (rootState: RootStateIF): boolean {
+      return (rootState.account.currentAccount.accountType === AccountTypes.PREMIUM)
+    },
+
+    /** Whether the user is ServiceBC Staff (which is not the same as Staff). */
+    isSbcStaff (rootState: RootStateIF): boolean {
+      return (rootState.account.currentAccount.accountType === AccountTypes.SBC_STAFF)
+    },
+
+    /** The user's Keycloak GUID. */
+    getKeycloakGuid (rootGetters): string {
+      return rootGetters['auth/keycloakGuid']
```

Comment:
Can you use dot notation here for the properties (like line 24) ?

Also on line 29.

Reply:
Better TS support using dot

Reply:
I'm having trouble changing this. So, `auth` doesn't have a `keycloakGuid` field. Hence, I can't access it via the dot notation. Same thing applies for line 29.
https://github.com/JazzarKarim/business-filings-ui/blob/fbb7af423f285df019befaacfd0f3fa7acf8c6e6/src/interfaces/root-state-interface.ts#L9

Reply:
That must be why Jonathan did it that way.

OK, just leave it.